doc/api/diagnostics_channel.md

```diff
@@ -0,0 +1,189 @@
+# Diagnostics Channel
+
```

Comment:
Purely cosmetic... Given the existing pattern (async_hooks, perf_hooks), using a consistent name `diagnostic_hooks` might be better for the module name

Reply:
Hmm...maybe. 🤔 

I just inherited the name from the original prototype so didn't really put much thought into that.

Reply:
I don't think `diagnostics_hooks` is a good name here as this module doesn't hook anything. There might be hooks added in e.g http,.. which use this module to publish data.
Actually it's not even limited to diagnostics data, it could be used for any sort of events.

Reply:
I was thinking more that the subscribers were "hooks" to intercept the message data of whatever named channels they are listening to. Seems like a reasonable name to me. Anyone else have any opinions? I'll leave the task of renaming for a bit to see what others think.